File: Day1/p1.py
#!/bin/python3

def readinput():
    with open("input") as f:
        return [int(line) for line in f]


def find2020p1(input):
    for idx in range(0, len(input) - 1):
        i = input[idx]
        for idx2 in range(idx+1, len(input)):
            j = input[idx2]
            if i + j == 2020:
                return i * j


def find2020p2(input):
    for idx in range(0, len(input) - 2):
        i = input[idx]
        for idx2 in range(idx + 1, len(input) - 1):
            j = input[idx2]
            if i + j > 2020:
                continue
            for idx3 in range(idx2 + 1, len(input)):
                k = input[idx3]
                if i + j + k == 2020:
                    return i * j * k

File: Day1/p1.go
package main

import (
	"bufio"
	"fmt"
//	"log"
	"os"
	"strconv"
)

func check(e error) {
    if e != nil {
        panic(e)
    }
}

func readfile() ([]int) {
	f, err := os.Open("input")
	defer f.Close()
	check(err)
	scanner := bufio.NewScanner(f)
	scanner.Split(bufio.ScanLines)
	var numbers []int
	for scanner.Scan() {
		i, err := strconv.Atoi(scanner.Text())
		check(err)
		numbers = append(numbers, i)
	}
	return numbers
}

func main() {
	numbers := readfile()
	count := len(numbers)
	for idx1, i := range numbers[0:count-2] {
		for idx2, j := range numbers[idx1 + 1:count-1] {
			if i + j == 2020 {
				fmt.Println(i * j)
			} else if (i + j < 2020) {
				for _, k := range numbers[idx2 + 1:count] {
					if i + j + k == 2020 {
						fmt.Println(i * j * k)
					}
				}
			}
		}
	}
}
